Sources/LiveKit/Support/ConnectivityListener.swift

```diff
@@ -0,0 +1,70 @@
+import Network
+
+internal protocol ConnectivityListenerDelegate: AnyObject {
+
+    func connectivityListener(_: ConnectivityListener, didUpdate hasConnectivity: Bool)
+    // network remains to have connectivity but path changed
+    func connectivityListener(_: ConnectivityListener, didSwitch path: NWPath)
+}
+
+internal extension ConnectivityListenerDelegate {
+    func connectivityListener(_: ConnectivityListener, didUpdate hasConnectivity: Bool) {}
+    func connectivityListener(_: ConnectivityListener, didSwitch path: NWPath) {}
+}
+
+internal extension NWPath {
+
+    func hasConnectivity() -> Bool {
+        if case .satisfied = status { return true }
+        return false
+    }
+}
+
+internal class ConnectivityListener: MulticastDelegate<ConnectivityListenerDelegate> {
+
+    static let shared = ConnectivityListener()
+
+    public private(set) var hasConnectivity: Bool {
+        didSet {
+            guard oldValue != hasConnectivity else { return }
+            notify { $0.connectivityListener(self, didUpdate: self.hasConnectivity) }
+        }
+    }
+
+    public private(set) var path: NWPath {
+        didSet {
+            defer { self.hasConnectivity = path.hasConnectivity() }
+
+            if oldValue.hasConnectivity(), path.hasConnectivity(),
+               // first interface changed
+               (oldValue.availableInterfaces.first != path.availableInterfaces.first)
```

Comment:
Is `first` the active one always?

Reply:
https://developer.apple.com/documentation/network/nwpath/2998720-availableinterfaces
"in order of preference" 

yes it should be !